numpy/f2py/cfuncs.py

```diff
@@ -1058,6 +1058,10 @@
                 tmp_fun = fun; /* built-in function */
                 Py_INCREF(tmp_fun);
                 tot = maxnofargs;
+                if (PyCFunction_Check(fun)) {
+                    /* In case the function has a co_argcount (like on PyPy) */
+                    di = 0;
```

Comment:
For other reviewers: this is used again on https://github.com/numpy/numpy/pull/15750/files#diff-57556f4d2e5b2a68017772da1827d14cR1100, although I don't really understand what's going on there.

Reply:
that link seems to point here?

Reply:
GitHub bug. It points to line 1100, but only if it's already expanded

Reply:
I think the intention is to reduce the argument count by 1 for methods, but rather than check "isinstance(method)" the check is "not a function". Since CFunctions (builtin functions) are not functions, but also do not have `co_argcount` on CPython, it works for CPython but fails for PyPy.

Reply:
If you've managed to work out what this `create_cb_arglist` function even does, can you add a comment on line 1034?

Reply:
Adding a comment to the best of my understanding.

Reply:
If you have an understanding of what any of the arguments mean, that would be great in a comment too.

Reply:
I don't really understand the code. It seems to have grown organically as a set of patches over the years, each time fixing one thing with no comments.

Reply:
Fair enough, let's leave this comment as is.